src/ecco/lm.py

```diff
@@ -209,7 +226,7 @@ def generate(self, input_str: str,
                  top_p: Optional[float] = None,
                  get_model_output: Optional[bool] = False,
                  do_sample: Optional[bool] = None,
-                 attribution: Optional[bool] = True,
+                 attribution: Optional[List[str]] = ['gradient', 'grad_x_input', 'integrated_gradients'],
```

Comment:
To improve performance of the base `generate` case (without attribution), I'm thinking we set the default to include none of the methods. On my laptop/CPU it takes IG about 3 seconds per token. The other methods run a bit faster (about 2-3 tokens per second). No attribution is about 10 tokens / second. 

On GPU, the speed is more reasonable (in that tokens stream to the screen quickly enough if you're expecting about 20 tokens.

What do you think about adding a short alias integrated gradients as a second option. So a user can specify either `attribution=['integrated_gradients']` or `attribution=['ig']` and either would work?

Reply:
> We can split the generation with attribution to a separate generate_with_attribution method, perhaps and change the behavior there

Instead of having 2 different methods, I prefer having `generate` with default `None` attributions as you suggest. 

> We can probably even switch all the methods to Captum (the existing saliency and input*grad) if that uses GPU directly and is faster. 

Using Captum for computing the saliency would make the code way cleaner and we would be able to use lm.model.generate for generating all the tokens (with beam search if needed! [we have a follow-up PR ready](https://github.com/JoaoLages/ecco/tree/feature-add-beam-search) 😁 ) instead of us generating the tokens one by one. However, it will make the code slightly slower for input saliency because you calculate the saliency scores on each generation step and Captum needs to perform the forward steps again in order to calculate the primary attributions.
Nonetheless, since the code will become way cleaner and we can make Captum work for a bunch of explainability methods, I think it's a good choice!

> And in that scenario showing the visualization right away would fit what the user expects. How does that sound?
Agreed, but maybe we can move the visualization to the `generate` method in another PR.

Reply:
Whoa adding beam search would be phenomenal!

Re: Captum for saliency. Yeah, that sounds like an okay trade-off. It likely should also make non-attribution generation easier since we can use the `past` parameter. Raw saliency is easy to compute but I wouldn't put too much faith in its scores. Something like IG is one of the better options currently as far as I understand.

Agreed on having a single generate method with `None` default in this PR and moving the visualization in an upcoming PR. 

Ready to merge when you advise. Anything else needed for this PR beyond changing the default?

Reply:
Cool, I'll make the following changes in this PR:

- use Captum to compute saliency;
- add short aliases like 'ig' for integrated gradients;
- set `None` as default attribution in `generate`.

Reply:
Done! Also introduced some new methods from Captum! 😄
Had to remove the attribution_test file as it didn't make sense anymore